bin/cylc-client

```diff
@@ -32,39 +32,35 @@ if '--use-ssh' in sys.argv[1:]:
         sys.exit(0)
 
 from cylc.option_parsers import CylcOptionParser as COP
-from cylc.network.httpclient import SuiteRuntimeServiceClient
+from cylc.network.client import SuiteRuntimeClient
 
 
 def main():
     parser = COP(__doc__, comms=True, argdoc=[
-        ('METHOD', 'Network API function name'),
-        ('[REG]', 'Suite name')])
+        ('REG', 'Suite name'),
+        ('METHOD', 'Network API function name')])
+
     parser.add_option(
         '-n', '--no-input',
         help='Do not read from STDIN, assume null input',
         action='store_true', dest='no_input')
     options, args = parser.parse_args()
-    # Client API function name
-    func = args[0]
-    if len(args) == 2:
-        # 2nd argument is suite name, assume authentication required
-        suite = args[1]
-        auth = None
-    else:
-        # No 2nd argument, assume anonymous authentication
-        suite = None
-        auth = SuiteRuntimeServiceClient.ANON_AUTH
-    pclient = SuiteRuntimeServiceClient(
-        suite, options.owner, options.host, options.port,
-        options.comms_timeout, my_uuid=options.set_uuid,
-        print_uuid=options.print_uuid, auth=auth)
+    suite = args[0]
+    func = args[1]
+    pclient = SuiteRuntimeClient(
+        suite, options.owner, options.host, options.port)
     if options.no_input:
         kwargs = {}
     else:
         kwargs = json.load(sys.stdin)
     sys.stdin.close()
-    sys.stdout.write(json.dumps(getattr(pclient, func)(**kwargs), indent=4))
+    sys.stdout.write(json.dumps(
+        pclient(func, kwargs, timeout=options.comms_timeout), indent=4
+    ) + '\n')
 
 
 if __name__ == '__main__':
-    main()
+    try:
+        main()
+    except Exception as exc:
+        sys.exit(exc)
```

Comment:
I think we now want the simpler style for all the other CLI - as users are much happier seeing Python tracebacks.

```python
if __name__ == '__main__':
    main()
```

Reply:
>  as users are much happier seeing Python tracebacks

**I wouldn't be so sure of that!**

Traceback scares a lot of users and conceals the information the user actually needs.

I'm in two minds over this one, Cylc's historic habit of concealing traceback has caused problems with cryptic error messages which fail to describe the issue.

A simple rule of thumb for traceback might go something like:

* If the user hits an "expected" error (e.g. typo) give them a nice informative one line error.
* If the user hits an unexpected issue traceback may be more appropriate.

Take the example where a user gets the command wrong. Which is best from a UX perspective?

```console
$ # catching the exception
$ cylc client generic ping -n
Request returned error: No method by the name "ping"

$ # raising the exception
$ cylc client generic ping -n
Traceback (most recent call last):
  File "/cylc/bin/cylc-client", line 63, in <module>
    main()
  File "/cylc/bin/cylc-client", line 58, in main
    pclient(func, kwargs, timeout=options.comms_timeout), indent=4
  File "/cylc/lib/cylc/network/client.py", line 194, in serial_request
    self.async_request(command, args, timeout))
  File "/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "/cylc/lib/cylc/network/client.py", line 190, in async_request
    raise ClientError(response['error'])
cylc.network.client.ClientError: Request returned error: No method by the name "ping"
```

Now imagine you have never written a line of Python in your life and don't understand the workings of Cylc.

How other tools handle the same scenario:
```console
$ ssh myhost lss
bash: lss: command not found

$ git statts
WARNING: You called a Git command named 'statts', which does not exist.

$ ls --reversse
ls: unrecognized option '--reversse'

$ cylc stop elephant
Contact info not found for suite "elephant", suite not running?
```

Two supplemental matters:
1. The cylc client commands are a bit unusual as in the event of a server side error we only really want the server side error not the full client stack.
2. On a sidenote in Python3 if you catch an exception and raise a new one in its place you will get both exceptions in your traceback, the error the user needs to see might not be at the bottom of the trace in this example.

What should we do to make the interface informative and simple?

* Create an exception type for expected errors
* Always output the full traceback then underneath output `str(exc)` in bold to make the error clear.
* ?

Reply:
If possible, all known errors should subclass `CylcError` or something like that - so we only catch `CylcError` and leave all other Exception un-caught.

Reply:
(May be another TODO.)

Reply:
> If possible, all known errors should subclass CylcError

Probably the most sensible approach, we could stick a decorator on the `main()` methods to handle this.

For the moment I've removed the try/except, will raise a follow-on issue.